app/scripts/modules/core/src/application/nav/NavCategory.spec.tsx

```diff
@@ -0,0 +1,127 @@
+import { Application, ApplicationModelBuilder } from '../../application';
+import { ApplicationDataSource, IDataSourceConfig } from '../service/applicationDataSource';
+import { IEntityTags, IServerGroup, IPipeline } from '../../domain';
+import { NavCategory } from './NavCategory';
+
+describe('NavCategory', () => {
+  const buildApp = <T,>(config: IDataSourceConfig<T>): Application =>
+    ApplicationModelBuilder.createApplicationForTests('testapp', config);
+
+  it('should render a datasources icon', () => {
+    const app = buildApp<IServerGroup>(mockServerGroupDataSourceConfig);
+    const category = app.getDataSource('serverGroups');
+    category.iconName = 'spMenuClusters';
+
+    const wrapper = mount(<NavCategory app={app} category={category} isActive={false} />);
+    const nodes = wrapper.children();
+    const icon = nodes.childAt(1).children();
+    expect(icon.find('svg').length).toEqual(1);
+  });
+
+  it('should render a placeholder when there is icon', () => {
+    const app = buildApp<IServerGroup>(mockServerGroupDataSourceConfig);
+    const category = app.getDataSource('serverGroups');
+
+    const wrapper = mount(<NavCategory app={app} category={category} isActive={false} />);
+    const nodes = wrapper.children();
+    const icon = nodes.childAt(1).children();
+    expect(icon.find('svg').length).toEqual(0);
+  });
+
+  it('should render running tasks badge', () => {
+    const app = buildApp<IPipeline>(mockPipelineDataSourceConfig);
+    const category = app.getDataSource('executions');
+    app.dataSources.push({ ...category, key: 'runningExecutions' } as ApplicationDataSource<IPipeline>);
+    app.getDataSource(category.badge).status$ = new BehaviorSubject({
+      status: 'FETCHED',
+      loaded: true,
+      lastRefresh: 0,
+      error: null,
+      data: [mockPipelineDataSourceConfig, mockPipelineDataSourceConfig],
+    });
+
+    const wrapper = mount(<NavCategory app={app} category={category} isActive={false} />);
+    const nodes = wrapper.children();
+    expect(nodes.find('.badge-running-count').length).toBe(1);
+    expect(nodes.find('.badge-none').length).toBe(0);
+
+    const text = nodes.childAt(0).getDOMNode();
+    expect(text.textContent).toBe('2');
+  });
+
+  it('should not render running tasks badge if there are none', () => {
+    const app = buildApp<IPipeline>(mockPipelineDataSourceConfig);
+    const category = app.getDataSource('executions');
+    app.dataSources.push({ ...category, key: 'runningExecutions' } as ApplicationDataSource<IPipeline>);
+
+    const wrapper = mount(<NavCategory app={app} category={category} isActive={false} />);
+    const nodes = wrapper.children();
+    expect(nodes.find('.badge-running-count').length).toBe(0);
+    expect(nodes.find('.badge-none').length).toBe(1);
+
+    const text = nodes.childAt(0).getDOMNode();
+    expect(text.textContent).toBe('');
+  });
+
+  it('subscribes to runningCount updates', () => {
```

Comment:
What is this testing?  Shouldn't it be asserting something related to runningCount?

Reply:
Yes, I forgot to push the update to this test. My bad. This is a an error on my part. It is meant to to push a running execution to the application and monitor the change in the runningCount.